spec/features/user_form_spec.rb

```diff
@@ -2,34 +2,75 @@
-    click_button("Save")
-    expect(page).to have_content("Success: User created successfully")
-
-    # Make sure password email url is correct and no missing translations
-    email = ActionMailer::Base.deliveries.last
-    expect(email.body.to_s).to match(%r{^https?://.+/en/password-resets/\w+/edit$})
-    expect(email.body.to_s).not_to match(/translation_missing/)
+  context "admin user in mission mode" do
+    scenario "create and edit user should work" do
+      visit("/en/m/#{mission.compact_name}/users/new")
+
+      # check for placeholders since they have not yet been overwritten
+      expect(page).to have_field("Main Phone", placeholder: "+17123241235")
+      expect(page).to have_field("Alternate Phone", placeholder: "+2348123456789")
+
+      fill_in("Full Name", with: "Foo Bar")
+      fill_in("Username", with: "foobar")
+      fill_in("Email", with: "foo@bar.com")
+      fill_in("Main Phone", with: "+17094554098")
+      fill_in("Alternate Phone", with: "+2347033772211")
+      select("Enumerator", from: "user_assignments_attributes_0_role")
+      select("Send password reset instructions via email", from: "user_reset_password_method")
+      click_button("Save")
+      expect(page).to have_content("Success: User created successfully")
+
+      # Make sure password email url is correct and no missing translations
+      email = ActionMailer::Base.deliveries.last
+      expect(email.body.to_s).to match(%r{^https?://.+/en/password-resets/\w+/edit$})
+      expect(email.body.to_s).not_to match(/translation_missing/)
+
+      # Go to show and see if correct
+      click_link("Foo Bar")
+      expect(page).to have_content("foo@bar.com")
+
+      # Go to edit and change something
+      click_link("Edit User")
+      expect(page).to have_content("Foo Bar")
+      select("Staffer", from: "user_assignments_attributes_0_role")
+      click_button("Save")
+
+      # Go to show page
+      click_link("View User")
+      expect(page).to have_content("Staffer")
+    end
+  end
+
+  context "admin user in admin mode" do
+    scenario "create and edit user should work" do
+      visit("/en/admin/users/new")
+      fill_in("Full Name", with: "Ella Baker")
+      fill_in("Username", with: "ellab")
+      fill_in("Email", with: "ella@baker.com")
+      find("a", text: "Add Assignment").click
+      select("staffer", from: "user[assignments_attributes][0][role]")
+      click_on("Save")
+      expect(page).to have_content("Success: User created successfully")
+
+      # Go to show user
+      click_link("Ella Baker")
+      expect(page).to have_content("Staffer")
+
+      # Edit user and change assignment, verify in show
+      click_link("Edit User")
+      select("reviewer", from: "user[assignments_attributes][0][role]")
+      click_on("Save")
+      expect(page).to have_content("Success: User updated successfully")
+      click_on("View User")
+      expect(page).to have_content("Reviewer")
```

Comment:
Also nice and clear, well done. You need to be careful with doing `have_content` on the whole page. You may get a false pass, if for instance 'Reviewer' appears somewhere else on the page, which is not totally unrealistic. You could perhaps try `"Role Reviewer"`, which would narrow it down a lot. Even better would be `have_css('.user_role', content: "Reviewer")`. Note this should change in several places in this file.